astropy/coordinates/earth.py

```diff
@@ -645,33 +645,47 @@ def get_gcrs_posvel(self, obstime):
             The GCRS velocity of the object
         """
         # GCRS position
-        gcrs_data = self._get_gcrs(obstime).data
+        gcrs_data = self.get_gcrs(obstime).data
         obsgeopos = gcrs_data.without_differentials()
         obsgeovel = gcrs_data.differentials['s'].to_cartesian()
         return obsgeopos, obsgeovel
 
-    def _gravitational_redshift(self, obstime):
+    _GM_moon = consts.G * 7.34767309e22*u.kg
+    def gravitational_redshift(self, obstime,
+                               bodies=('sun', 'jupiter', 'moon', 'earth'),
+                               masses=(consts.GM_sun, consts.GM_jup,
+                                       _GM_moon, consts.GM_earth)):
         """Return the gravitational redshift at this EarthLocation.
 
-        Calculates the gravitational redshift, of order 3 m/s, due to the Sun,
-        Jupiter, the Moon, and the Earth itself.
+        Calculates the gravitational redshift, of order 3 m/s, due to the
+        requested solar system bodies.
 
         Parameters
         ----------
         obstime : `~astropy.time.Time`
             The ``obstime`` to calculate the redshift at.
 
+        bodies : list
+            The bodies to include in the redshift calculation.  List elements
+            should be any body name `get_body_barycentric` accepts.  Defaults to
+            Jupiter, the Sun, the Moon, and Earth Itself.
+
+        masses : list of Quantity
+            The gravitational parameters (G * mass) to assume for the bodies
+            requested in ``bodies``. Length must match ``bodies``.
+
         Returns
         --------
         redshift :  `~astropy.units.Quantity`
             Gravitational redshift in velocity units at given obstime.
         """
         # needs to be here to avoid circular imports
         from .solar_system import get_body_barycentric
-        names = ('sun', 'jupiter', 'moon', 'earth')
-        GM_moon = consts.G * 7.34767309e22*u.kg
-        masses = (consts.GM_sun, consts.GM_jup, GM_moon, consts.GM_earth)
-        positions = [get_body_barycentric(name, obstime) for name in names]
+
+        if len(masses) != len(bodies):
+            raise ValueError("masses and bodies must be same length")
```

Comment:
I'm tempted to add more checks here, or perhaps use the ```quantity_input``` decorator for the ```masses``` argument. If the user gets the units of one wrong in the list the error message doesn't help them know which, but perhaps more worryingly is the case where they get all the units wrong.